src/main.rs

```diff
@@ -47,125 +87,249 @@ impl LogLevel {
-    ] {
-        for entry in glob(path)?.flatten() {
+        path_exists = true;
+        let greenboot_required_path = format!("{greenboot_required_path}*.sh");
+        for entry in glob(&greenboot_required_path)?.flatten() {
             log::info!("running required check {}", entry.to_string_lossy());
-            let output = Command::new("bash").arg("-C").arg(entry).output()?;
+            let output = Command::new("bash").arg("-C").arg(entry.clone()).output()?;
             if !output.status.success() {
-                // combine and print stderr/stdout
-                log::warn!("required script failed...");
-                failure = true;
+                log::error!("required script {} failed!", entry.to_string_lossy());
+                log::error!("reason: {}", String::from_utf8_lossy(&output.stderr));
+                script_failure = true;
             }
         }
     }
-    for path in [
-        "/usr/lib/greenboot/check/wanted.d/*.sh",
-        "/etc/greenboot/check/wanted.d/*.sh",
-    ] {
-        for entry in glob(path)?.flatten() {
-            log::info!("running required check {}", entry.to_string_lossy());
-            let output = Command::new("bash").arg("-C").arg(entry).output()?;
+
+    if !path_exists {
+        bail!("required.d not found");
+    }
+
+    for path in GREENBOOT_INSTALL_PATHS {
+        let gereenboot_wanted_path = format!("{path}/check/wanted.d/*.sh");
+        for entry in glob(&gereenboot_wanted_path)?.flatten() {
+            log::info!("running wanted check {}", entry.to_string_lossy());
+            let output = Command::new("bash").arg("-C").arg(entry.clone()).output()?;
             if !output.status.success() {
                 // combine and print stderr/stdout
-                log::warn!("wanted script failed...");
+                log::warn!("wanted script {} failed!", entry.to_string_lossy());
+                log::warn!("reason: {}", String::from_utf8_lossy(&output.stderr));
             }
         }
     }
-    // if a command with restart option in systemd fails to start we don't get it as "failed"
-    // reversing the check makes sure that if by the time After=multi-user the service isn't running then it's failing at least
-    let output = Command::new("systemctl")
-        .arg("list-units")
-        .arg("--state")
-        .arg("active")
-        .arg("--no-page")
-        .arg("--output")
-        .arg("json")
-        .output()?;
-    let services: Vec<ServiceStatus> = serde_json::from_str(&String::from_utf8(output.stdout)?)?;
-    let ss: Vec<String> = services.iter().map(|x| x.unit.clone()).collect();
-    let active_units: HashSet<String> = HashSet::from_iter(ss);
-    for service in ["sshd.service", "NetworkManager.service"] {
-        if !active_units.contains(service) {
-            log::warn!("service {} failed, see journal", service);
-            failure = true;
-        }
+
+    if script_failure {
+        bail!("health-check failed!");
```

Comment:
Which health check failed?

Reply:
I think that Peter means that you should write a more descriptive error

Reply:
we are logging things extensively, lets take this a a separate issue to fix logging.